### PR TITLE
[Helm] Consolidate Helm chart sets

### PR DESCRIPTION
**Pull request by CyrineG**
Signed-off-by: CyrineG <syrynegamoudi@gmail.com>

**Description**

This PR fixes https://github.com/meshery/meshery/issues/5959

The objective  is to consolidate the duplicated Helm Chart sets under [meshery/install/helm/meshery/charts](https://github.com/meshery/meshery/tree/master/install/kubernetes/helm/meshery/charts) and [meshery/install/helm/meshery-operator/charts](https://github.com/meshery/meshery/tree/master/install/kubernetes/helm/meshery-operator/charts)

**Notes for Reviewers**
The number of changes look scary, (please don't run away :laughing: )  but I just did this:

1. Following the [Helm documentation](https://helm.sh/docs/helm/helm_dependency/?fbclid=IwAR00c5QS-kNHZeE7KmWRizhhyky-12OnaIItSpiveDQlzEsU4Vg6K2ZJ9xE), I changed `/helm/meshery/Chart.yaml` so its dependency charts have as path, the path of the charts stored locally under helm/meshery-operator . 
2. Run `helm dep update`. This created `/helm/meshery/Chart.lock` and put subcharts (*.tgz files) into the `./charts` folder 

I believe that this solves the Chart sets duplication issue since, now there is a unique chart set under helm/meshery-operator path. Any change to them should be followed by running `helm dep update`.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@CyrineG, if you haven't seen the two sequence diagrams in the Meshery Architecture deck, I encourage you to take look as they might offer some clarity; might reinforce your approach.

**Comment by Revolyssup** [comment]
@CyrineG  In this PR, meshery-operator's Chart.yaml has adapter's charts in dependencies. This should not be the case. 
Meshery server or any of Meshery adapters like `meshery-istio` , `meshery-linkerd` should not be dependencies of meshery-operator chart. The only dependencies `meshery-operator` has are: broker and meshsync.

In terms of dependencies across charts and subscharts, here is the basic idea:
Meshery's chart has all adapters as well as meshery operator as dependencies and enabled as true. 

Then Meshery-operator has only two dependencies, meshsync and broker.
When we want to install just the operator, we should be able to use operator chart.

**Comment by CyrineG** [comment]
@Revolyssup done!

**Comment by Revolyssup** [comment]
> @Revolyssup done!

Have you tried installing these charts in different flows? I will checkout your branch and perform the below tests. Meanwhile if you get time before me, make sure we can:
1. Install Meshery with or without operator by some override, having adapters as dependencies.
2. Install Meshery operator alone.

**Comment by CyrineG** [comment]
@Revolyssup so here's what I tried:
`> helm install meshery meshery/. --set meshery-operator.enabled=false --set meshery-broker.enabled=false --set meshery-meshsync.enabled=false`

`>  kubectl get pods -n meshery`

      NAME                                   READY   STATUS             RESTARTS     AGE
      meshery-5d8d749fb4-fj9xk               1/1     Running            0            32s
      meshery-app-mesh-67c78f557d-99vm5      1/1     Running            0            32s
      meshery-cilium-7757c94c4d-4b59n        0/1     CrashLoopBackOff   1 (7s ago)   32s
      meshery-consul-8484fbcc96-b9795        1/1     Running            0            32s
      meshery-cpx-7c6885bcfc-tcr6m           1/1     Running            0            32s
      meshery-istio-65f67fb978-95nqt         1/1     Running            0            32s
      meshery-kuma-785765997-qvvp8           1/1     Running            0            32s
      meshery-linkerd-cc65f9d78-f9vhv        1/1     Running            0            31s
      meshery-nginx-sm-5c5569dbf4-2h6mf      1/1     Running            0            32s
      meshery-nsm-7788546ff6-lmbq8           1/1     Running            0            32s
      meshery-osm-6fbc9769cd-vwtxn           1/1     Running            0            32s
      meshery-perf-794bb4f4c7-hknf9          0/1     CrashLoopBackOff   1 (6s ago)   32s
      meshery-traefik-mesh-c5548d4f6-7wkfl   1/1     Running            0            32s

`> kubectl logs meshery-5d8d749fb4-fj9xk -n meshery`

      time="2022-08-17T11:47:59Z" level=info msg="Local Provider capabilities are: v0.6.0-rc.6fd"
      time="2022-08-17T11:47:59Z" level=info msg="Meshery Server release channel is: stable"
      time="2022-08-17T11:47:59Z" level=info msg="Meshery Database is at: /home/appuser/.meshery/config"
      time="2022-08-17T11:47:59Z" level=info msg="Using kubeconfig at: /home/appuser/.kube"
      time="2022-08-17T11:47:59Z" level=info msg="Log level: info"
      time="2022-08-17T11:47:59Z" level=info msg="creating directories for seeding... /home/appuser/.meshery/content/patterns" app=meshery
      time="2022-08-17T11:47:59Z" level=info msg="creating directories for seeding... /home/appuser/.meshery/content/applications" app=meshery
     ....
      time="2022-08-17T11:47:59Z" level=info msg="seeding sample Filters" app=meshery
      time="2022-08-17T11:48:00Z" level=info msg="extracting Applications from /home/appuser/.meshery/content/applications" app=meshery
      time="2022-08-17T11:48:00Z" level=info msg="seeding sample Applications" app=meshery
      time="2022-08-17T11:48:00Z" level=info msg="Meshery Server listening on :8080"

`> helm install meshery-operator meshery-operator/. --namespace=meshery`

    Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: MeshSync "meshery-meshsync" in namespace "meshery" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "meshery-operator": current value is "meshery"

From what I understand, this means that meshery-meshsync is being created with meshery hence why it has release-name=meshery. But it's not being created at that point??

**Comment by Revolyssup** [comment]
Try putting meshery-meshsync and meshery-broker charts only in meshery-operator chart. And in Meshery chart, make just meshery-operator as a local dependency.

**Comment by CyrineG** [comment]
@Revolyssup I'm sorry for taking long to respond, unfortunately I won't be able to look into this till after next week.  I'll definitely come back to this once I'm available again.

**Comment by leecalcote** [comment]
@acald-creator, your review here would be most helpful.

**Comment by CyrineG** [comment]
Both the Release Charts test s is failing because Helm expects all dependencies (meshery-cpx, meshery-perf included) that exist under charts/ to be declared in Chart.yaml :thinking:

**Comment by alphaX86** [comment]
> Both the Release Charts tests are failing here because Helm expects all dependencies (meshery-cpx, meshery-perf included) that exist under charts/ to be declared in Chart.yaml 🤔

Right now, I raised revert PR #6180 to fix this issue (needs a release to reflect it as well) but we need to find the root cause on why it's failing even though I removed them from the list in `Charts.yaml`

**Comment by acald-creator** [comment]
I'm testing to see if it made a difference.

**Comment by Revolyssup** [comment]
@CyrineG  Can you solve the merge conflicts?

**Comment by CyrineG** [comment]
@Revolyssup done!

File: install/kubernetes/helm/meshery/Chart.yaml [hunk 1]
```diff
@@ -75,14 +75,9 @@ dependencies:
...
-    version: 0.5.0
+    version: 0.6.0
     condition: meshery-operator.enabled
-  - name: meshery-broker
-    version: 0.5.0
-    condition: meshery-broker.enabled
-  - name: meshery-meshsync
-    version: 0.5.0
-    condition: meshery-meshsync.enabled
+    repository: "file://../meshery-operator"
```
Comment (alphaX86) [inline install/kubernetes/helm/meshery/Chart.yaml:80]: Is there any other way we can reference the repo with Artifact hub link?

Comment (leecalcote) [reply install/kubernetes/helm/meshery/Chart.yaml:80]: The official repo for published Meshery Charts is https://meshery.io/charts. Artifact Hub points people to this same repo.

**Comment by Revolyssup** [comment]
@CyrineG  I am not able to checkout this PR. 
`From github.com:meshery/meshery
 ! [rejected]            refs/pull/6005/head -> refactor/cyrineg/5959  (non-fast-forward)
exit status 1
`

**Comment by acald-creator** [comment]
Helm deployment for normal `meshery` and `meshery-operator` alone works for me.

**Comment by Revolyssup** [comment]
@CyrineG  So by default meshery chart installs operator. The only caviat is, when operator is installed with `meshery` chart then it comes under the release name `meshery`. But when users try to uninstall the operator, it looks up specifically for release name `meshery-operator` and when its not found then uninstall fails.

**Comment by CyrineG** [comment]
@Revolyssup Looking at [this](https://codeutility.org/kubernetes-where-to-override-release-name-in-helm-stack-overflow/) it  doesn't seem possible for a subchart to override the release name that was defined by `Helm install`. The same happens when I deploy meshery with operator using the old set of charts, only the `meshery` release exists so it seems weird that this wasn't an issue with old charts too 🤔 

How are you testing the uninstallation, using the UI?

**Comment by Revolyssup** [comment]
> @Revolyssup Looking at [this](https://codeutility.org/kubernetes-where-to-override-release-name-in-helm-stack-overflow/) it doesn't seem possible for a subchart to override the release name that was defined by `Helm install`. The same happens when I deploy meshery with operator using the old set of charts, only the `meshery` release exists so it seems weird that this wasn't an issue with old charts too 🤔
> 
> How are you testing the uninstallation, using the UI?

On the settings page, I click on the kubernetes context and on the right side there is a toggle button to turn off operator.

**Comment by leecalcote** [comment]
Thank you, @CyrineG

**Pull request closed without merging**